TestingHomework.Tests/Tests/ProductAdminManagerTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+using TestingHomework_Discounts.Managers;
+using TestingHomework_Discounts;
+using System.Linq;
+using TestingHomework.Tests.Data;
+using TestingHomework.Tests.DataPrep;
+using NUnit.Framework;
+using TestingHomework_Discounts.Models;
+using DeepEqual.Syntax;
+using TestingHomework.Tests.TestBases;
+using Moq;
+using Telerik.JustMock.AutoMock;
+using TestingHomework.Tests.Accessors;
+
+namespace TestingHomework.Tests
+{
+    public class ProductAdminManagerTests: ManagerTestBase
+    {
+        MockingContainer<ProductAdminManager> mockContainer = new MockingContainer<ProductAdminManager>();
+
+        ProductAdminManager productAdminManager;
+ 
+        public override void OnInitialize()
+        {
+
+            productAdminManager = mockContainer.Instance;
+        }
+
+        [Test]
+        public void SaveProduct()
+        {
+            var product = dataPrep.Products.Create();
```

Comment:
These tests do use dataprep, deep equal, and mocking correctly. However, the tests have become trivial because all of the logic has been moved to the accessor. The accessors are not tested and so none of the application's logic is actually verified.